### PR TITLE
replace mkdir with makedirs so that it works even if you do not yet h…

### PR DESCRIPTION
**Pull request by jshipton**
…ave a results directory

### CODE REVIEW

**Comment by jshipton** [comment]
@thomasgibson Upon using a clean installation of Gusto I just discovered that `mkdir` is not recursive which means that if you haven't run yet and therefore don't have a `results` directory, it crashes. This tiny change should fix that...

File: gusto/state.py [hunk 1]
```diff
@@ -333,7 +333,7 @@ def setup_dump(self, t, tmax, pickup=False):
                               % self.dumpdir)
             else:
                 if not running_tests:
-                    mkdir(self.dumpdir)
+                    makedirs(self.dumpdir)
```
[the remark this answers is not in the document]
Comment (jshipton) [reply gusto/state.py]: It seems to be working how I'd like it to but I'm not quite sure what you mean... `self.dumpdir` is made by joining `results/` with another path. This errors when the full path exists but seems happy to create `results/newdir` even if `results/` already exists. That's what I want.

[the remark this answers is not in the document]
Comment (wence-) [reply gusto/state.py]: Yes, so if you want:

```python
os.makedirs("/already/existing/dump/directory")
```
to error, then you have things right.

If you want:

```python
os.makedirs("/already/existing/dump/directory")
```
to succeed, you need to say `os.makedirs(..., exist_ok=True)`

[the remark this answers is not in the document]
Comment (jshipton) [reply gusto/state.py]: The bit I was confused by was whether it cares if intermediate directories exists - e.g. does it care if `"already/existing"` already exists. It seems not and I still want it to error when the entire path exists - causes no end of irritation but nothing compared to accidentally overwriting a ton of results.

**Pull request merged**